dvc/progress.py

```diff
@@ -61,15 +68,24 @@ def __init__(
                 logging.getLogger(__name__).getEffectiveLevel()
                 >= logging.CRITICAL
             )
+        if bar_format is None:
+            if kwargs.get('total', getattr(iterable, '__len__', None)) is None:
```

Comment:
```suggestion
            if kwargs.get('total', hasattr(iterable, '__len__')):
```

Reply:
Also, shouldn't we move to constructor? or to whatever code that calls `len()`?

Reply:
`super()` hasn't been called yet so `len()` won't work. And `super()` shouldn't be called before `bar_format` is defined. A work-around would be to set `bar_format = ""`, call `super()`, then `len()`.

Reply:
also the suggested patch doesn't work - it needs to be inverted :)

Reply:
delayed `bar_format` now